pc/storage/kdtree/kdtree_test.go

```diff
@@ -198,28 +467,352 @@ func TestKDtree(t *testing.T) {
+					hasError: false,
+					expectedTree: &KDTree{
+						Vec3RandomAccessor: it,
+						root: &node{
+							children: [2]*node{
+								&node{
+									children: [2]*node{
+										&node{
+											id:  5,
+											dim: 2,
+										},
+										&node{
+											id:  1,
+											dim: 2,
+										},
+									},
+									id:  4,
+									dim: 1,
+								},
+								&node{
+									id:  2,
+									dim: 1,
+								},
+							},
+							id:  0,
+							dim: 0,
+						},
+					},
+				},
+			},
+			"NodeWithBothLeftAndRightSubTrees": {
+				{
+					pID:      0,
+					hasError: false,
+					expectedTree: &KDTree{
+						Vec3RandomAccessor: it,
+						root: &node{
+							children: [2]*node{
+								&node{
+									children: [2]*node{
+										&node{id: 5, dim: 2},
+										&node{id: 1, dim: 2},
+									},
+									id:  4,
+									dim: 1,
+								},
+								&node{
+									children: [2]*node{
+										&node{id: 2, dim: 2},
+									},
+									id:  6,
+									dim: 1,
+								},
+							},
+							id:  3,
+							dim: 0,
+						},
+					},
+				},
+			},
+			"TwiceTheSamePoint": {
+				{
+					pID:      3,
+					hasError: false,
+					expectedTree: &KDTree{
+						Vec3RandomAccessor: it,
+						root: &node{
+							children: [2]*node{
+								&node{
+									children: [2]*node{
+										&node{
+											id:  5,
+											dim: 2,
+										},
+										&node{
+											id:  1,
+											dim: 2,
+										},
+									},
+									id:  4,
+									dim: 1,
+								},
+								&node{
+									children: [2]*node{
+										&node{
+											id:  2,
+											dim: 2,
+										},
+									},
+									id:  6,
+									dim: 1,
+								},
+							},
+							id:  0,
+							dim: 0,
+						},
+					},
+				},
+				{
+					pID:      3,
+					hasError: false,
+					expectedTree: &KDTree{
+						Vec3RandomAccessor: it,
+						root: &node{
+							children: [2]*node{
+								&node{
+									children: [2]*node{
+										&node{
+											id:  5,
+											dim: 2,
+										},
+										&node{
+											id:  1,
+											dim: 2,
+										},
+									},
+									id:  4,
+									dim: 1,
+								},
+								&node{
+									children: [2]*node{
+										&node{
+											id:  2,
+											dim: 2,
+										},
+									},
+									id:  6,
+									dim: 1,
+								},
+							},
+							id:  0,
+							dim: 0,
+						},
+					},
+				},
+			},
+			"InvalidPointID": {
+				{
+					pID:      -1,
+					hasError: true,
+				},
+				{
+					pID:      123,
+					hasError: true,
+				},
+			},
+		}
+		for name, steps := range testCases {
+			steps := steps
+			t.Run(name, func(t *testing.T) {
+				kdt = New(it)
+				for _, tt := range steps {
+					err := kdt.DeletePoint(tt.pID)
+					testKDT := tt.expectedTree
+					if tt.hasError {
+						if err == nil {
+							t.Errorf("Expected an error when trying to delete a point that is not in the tree")
+						}
+						testKDT = kdt
```

Comment:
I guess you want `t.FailNow()` instead of `testKDT = kdt`

Reply:
The idea was to test that when an error occurs, the tree remained unmodified.

Reply:
But this just compare the same pointer. It's always true even if the contents are changed.

Reply:
I made error cases similar as to the other ones in 4b4578afc89de4a9ca45369603fc0b09cba127c9.